interactions/api/gateway/client.py

```diff
@@ -401,37 +401,76 @@ def _dispatch_event(self, event: str, data: dict) -> None:
                 data["_client"] = self._http
                 obj = model(**data)
 
+                _cache: "Storage" = self._http.cache[model]
+
+                if isinstance(obj, Member):
+                    id = (Snowflake(data["guild_id"]), obj.id)
+                else:
+                    id = getattr(obj, "id", None)
+
+                if id is None:
+                    if model.__name__.startswith("Guild"):
+                        if model.__name__ == "GuildScheduledEventUser":
+                            id = model.guild_scheduled_event_id
+                        elif model.__name__ in [
+                            "Invite",
+                            "GuildBan",
+                            "ChannelPins",
+                            "MessageReaction",
+                            "ReactionRemove",
+                            # Extend this for everything that should not be cached
```

Comment:
Instead of doing that, wouldn't it be better to make a list of things that should be cached?
Maybe add it as a property to Cache? idk

Reply:
There is in general not a problem with caching everything, we just have to especially exclude the gw events that don't provide an ID

Reply:
Oh ok